test/suites/integration/grpcroute_test.go

```diff
@@ -0,0 +1,371 @@
+						},
+					},
+				},
+			},
+		})
+
+		testFramework.ExpectCreated(ctx, grpcRoute)
+
+		var tgSummary = testFramework.GetTargetGroupWithProtocol(ctx, grpcBinService, "http", "grpc")
+		tg, err := testFramework.LatticeClient.GetTargetGroup(&vpclattice.GetTargetGroupInput{TargetGroupIdentifier: tgSummary.Id})
+		Expect(err).To(BeNil())
+		Expect(*tg.Config.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttp))
+		Expect(*tg.Config.ProtocolVersion).To(Equal(vpclattice.TargetGroupProtocolVersionGrpc))
+		route, _ := core.NewRoute(grpcRoute)
+		latticeService = testFramework.GetVpcLatticeService(ctx, route)
+		Eventually(func(g Gomega) {
+			rules, err := testFramework.GetLatticeServiceHttpsListenerNonDefaultRules(ctx, latticeService)
+			g.Expect(err).To(BeNil())
+			g.Expect(len(rules)).To(Equal(1))
+			g.Expect(*rules[0].Match.HttpMatch.Method).To(Equal("POST"))
+			g.Expect(*rules[0].Match.HttpMatch.PathMatch.Match.Prefix).To(Equal("/"))
+		}).Should(Succeed())
+
+		grpcurlCmdOptions := test.RunGrpcurlCmdOptions{
+			GrpcServerHostName:  *latticeService.DnsEntry.DomainName,
+			GrpcServerPort:      "443",
+			Service:             "grpcbin.GRPCBin",
+			Method:              "DummyUnary",
+			ReqParamsJsonString: `{ "f_string": "myTestString", "f_int32": 42, "f_bytes": "SGVsbG8gV29ybGQ="}`,
+			UseTLS:              true,
+		}
+
+		Eventually(func(g Gomega) {
+			stdoutStr, stderrStr, err := testFramework.RunGrpcurlCmd(grpcurlCmdOptions)
+			g.Expect(err).To(BeNil())
+			g.Expect(stderrStr).To(BeEmpty())
+			g.Expect(stdoutStr).To(ContainSubstring(`"fString": "myTestString"`))
+			g.Expect(stdoutStr).To(ContainSubstring(`"fInt32": 42`))
+		}).Should(Succeed())
+
+		grpcurlCmdOptions = test.RunGrpcurlCmdOptions{
+			GrpcServerHostName:  *latticeService.DnsEntry.DomainName,
+			GrpcServerPort:      "443",
+			Service:             "addsvc.Add",
+			Method:              "Sum",
+			ReqParamsJsonString: `{"a": 5, "b": 6}`,
+			UseTLS:              true,
+		}
+		//Happy path: Verify client is able to invoke addsvc.Add/Sum method
+		Eventually(func(g Gomega) {
+			stdoutStr, stderrStr, err := testFramework.RunGrpcurlCmd(grpcurlCmdOptions)
+			g.Expect(err).To(BeNil())
+			g.Expect(stderrStr).To(BeEmpty())
+			g.Expect(stdoutStr).To(ContainSubstring("\"v\": " + strconv.Itoa(5+6)))
+		}).Should(Succeed())
+
+		grpcurlCmdOptions = test.RunGrpcurlCmdOptions{
+			GrpcServerHostName:  *latticeService.DnsEntry.DomainName,
+			GrpcServerPort:      "443",
+			Service:             "addsvc.Add",
+			Method:              "Concat",
+			ReqParamsJsonString: `{ "a": "Str1", "b": "Str2"}`,
+			UseTLS:              true,
+		}
+		//Happy path: Verify client is able to invoke addsvc.Add/Concat method
+		Eventually(func(g Gomega) {
+			stdoutStr, stderrStr, err := testFramework.RunGrpcurlCmd(grpcurlCmdOptions)
+			g.Expect(err).To(BeNil())
+			g.Expect(stderrStr).To(BeEmpty())
+			g.Expect(stdoutStr).To(ContainSubstring(`"v": "Str1Str2"`))
+		}).Should(Succeed())
```

Comment:
any plans to split each traffic test into "It" and global test into "Context"? Any guidance from https://onsi.github.io/ginkgo/#writing-specs?

Reply:
Changed in my new revision, please take a look